File: src/typescript/structure/organisation/dead-code/OrderProcessor.ts
import { readFileSync } from 'fs';

/**
 * This function calculates the discount for a user.
 * (Note: Currently, it doesn't apply any discount.)
 */
function calculateDiscount(amount: number): number {
    const discountRate = 0.1;
    return amount * discountRate;
}

class OrderProcessor {
    private orders: Order[] = [];
    private taxRate: number = 0.07;

    private errorMessage: string = "Order was not processed";

    constructor(private logger: Logger) {}

    processOrder(order: Order): void {
        this.orders.push(order);
        this.logger.log(`Processing order ID: ${order.id}`);

        if (true) {
            this.applyTax(order);
        } else {
            this.applyDiscount(order);
        }

        if (false) {
            this.sendConfirmationEmail(order);
        }

        const processingTime: number = this.calculateProcessingTime(order);
        this.logger.log(`Processing time for order ID ${order.id}: ${processingTime}ms`);
    }

    private applyTax(order: Order): void {
        order.total += order.total * this.taxRate;
        this.logger.log(`Applied tax to order ID: ${order.id}`);
    }

    private applyDiscount(order: Order): void {
        const discount = calculateDiscount(order.total);
        order.total -= discount;
        this.logger.log(`Applied discount to order ID: ${order.id}`);
    }

    private sendConfirmationEmail(order: Order): void {
        this.logger.log(`Sent confirmation email for order ID: ${order.id}`);
    }

    private calculateProcessingTime(order: Order): number {
        return 100;
    }
}

interface Order {
    id: number;
    total: number;
}

class Logger {
    log(message: string): void {
        console.log(message);
    }
}

const logger = new Logger();
const processor = new OrderProcessor(logger);

const newOrder: Order = { id: 1, total: 200 };
processor.processOrder(newOrder);
\ No newline at end of file

File: truth_set/go/structure/organisation/dead-code/OrderProcessor.go
package orderprocessor

import (
	"fmt"
)

func calculateDiscount(amount float64) float64 {
	discountRate := 0.1
	return amount * discountRate
}

type Order struct {
	ID    int
	Total float64
}

type Logger struct{}

func (l *Logger) Log(message string) {
	fmt.Println(message)
}

type OrderProcessor struct {
	orders       []Order
	taxRate      float64
	errorMessage string
	logger       *Logger
}

func NewOrderProcessor(logger *Logger) *OrderProcessor {
	return &OrderProcessor{
		orders:       make([]Order, 0),
		taxRate:      0.07,
		errorMessage: "Order was not processed",
		logger:       logger,
	}
}

func (p *OrderProcessor) ProcessOrder(order Order) {
	p.orders = append(p.orders, order)
	p.logger.Log(fmt.Sprintf("Processing order ID: %d", order.ID))

	if true {
		p.applyTax(&order)
	} else {
		p.applyDiscount(&order)
	}

	if false {
		p.sendConfirmationEmail(&order)
	}
}

func (p *OrderProcessor) applyTax(order *Order) {
	order.Total += order.Total * p.taxRate
	p.logger.Log(fmt.Sprintf("Applied tax to order ID: %d", order.ID))
}

func (p *OrderProcessor) applyDiscount(order *Order) {
	discount := calculateDiscount(order.Total)
	order.Total -= discount
	p.logger.Log(fmt.Sprintf("Applied discount to order ID: %d", order.ID))
}

func (p *OrderProcessor) sendConfirmationEmail(order *Order) {
	p.logger.Log(fmt.Sprintf("Sent confirmation email for order ID: %d", order.ID))
}

func main() {
	logger := &Logger{}
	processor := NewOrderProcessor(logger)

	newOrder := Order{ID: 1, Total: 200}
	processor.ProcessOrder(newOrder)
}
\ No newline at end of file